metals/src/main/scala/scala/meta/internal/pantsbuild/BloopPants.scala

```diff
@@ -538,69 +503,58 @@ private class BloopPants(
     )
   }
 
-  private val exportClasspathCache: mutable.Map[String, List[Path]] =
-    mutable.Map.empty[String, List[Path]]
-  private val exportClasspathDir: AbsolutePath = AbsolutePath(
-    workspace.resolve("dist").resolve("export-classpath")
-  )
-  private val exportClasspathJars = mutable.Map.empty[Path, Option[Path]]
-  def exportClasspathJar(
-      target: PantsTarget,
-      entry: Path,
-      suffix: String
-  ): Option[Path] = {
-    exportClasspathJars.getOrElseUpdate(
-      entry, {
-        val out = bloopJars.resolve(s"${target.id}$suffix.jar")
-        try {
-          val attr = Files.readAttributes(entry, classOf[BasicFileAttributes])
-          if (attr.isRegularFile()) {
-            // Copy jar from `.pants.d/` directory to `.bloop/` directory to ensure that the
-            // Bloop classpath is isolated from the Pants classpath.
-            Files.copy(entry, out, StandardCopyOption.REPLACE_EXISTING)
-            Some(out)
-          } else {
-            // Leave directory entries unchanged, don't copy them to `.bloop/` directory.
-            Some(entry)
-          }
-        } catch {
-          case _: IOException =>
-            // Does not exist, ignore this entry.
-            None
-        }
-      }
-    )
+  private def toImmutableJar(library: PantsLibrary, path: Path): Path = {
+    val filename = BloopPants.makeReadableFilename(library.name) + ".jar"
+    toImmutableJar(filename, path)
   }
-
-  private val bloopJars =
-    Files.createDirectories(bloopDir.resolve("bloop-jars"))
-  private def exportClasspath(target: PantsTarget): List[Path] = {
-    exportClasspathCache.getOrElseUpdate(
-      target.name, {
-        val classpathFile =
-          exportClasspathDir.resolve(target.id + "-classpath.txt")
-        if (classpathFile.isFile) {
-          val classpath =
-            Classpath(classpathFile.readText.trim()).entries.map(_.toNIO)
-          classpath.iterator.zipWithIndex.flatMap {
-            case (entry, 0) => exportClasspathJar(target, entry, "")
-            case (entry, i) => exportClasspathJar(target, entry, s"-$i")
-          }.toList
-        } else {
-          Nil
+  private def toImmutableJar(filename: String, path: Path): Path = {
+    // NOTE(olafur): Jars that live inside $WORKSPACE/.pants.d get overwritten
+    // by Pants during compilation. We copy these jars over to the Bloop
+    // directory so that the Bloop incremental cache is unaffected by the
+    // `./pants compile` tasks that the user is running.
+    if (path.startsWith(mutableJarsHome)) {
```

Comment:
Just found an issue because this method doesn't handle directory entries. Will need to fix before merge